pkg/controllers/user/alert/deployer/upgradeimpl.go

```diff
@@ -121,6 +121,10 @@ func (l *AlertService) Upgrade(currentVersion string) (string, error) {
 		if err := l.removeLegacyAlerting(); err != nil {
 			return "", err
 		}
+
+		if err := l.removeFinalizerFromLegacyAlerting(); err != nil {
```

Comment:
Should we remove those old alerts' finalizer no matter the alert version is latest or not? I think we should use a handler for them instead of removing them in upgrade logic.

Reply:
Just move this logic out of `strings.Contains` and it will be fine.

Reply:
updated.